src/common/mem.c

```diff
@@ -34,7 +34,13 @@ struct umem_tx_stage_item {
 
 #ifdef DAOS_PMEM_BUILD
 
-static int use_bmem = 1; /* Use BMEM by default */
+enum {
+	DAOS_MD_PMEM	= 0,
+	DAOS_MD_BMEM	= 1,
+	DAOS_MD_ADMEM	= 2,
+};
+
+static int daos_md_backend = DAOS_MD_BMEM; /* Use BMEM by default */
```

Comment:
This comment looks confusing, DAOS_MD_ON_SSD isn't set by default, so by default, the daos_md_backend will actually be set to DAOS_MD_PMEM in umempobj_setting_init().

If we want to use PMEM as default value, we need to fix this comment; If we want to use BMEM as default value, the umempobj_setting_init() should be fixed.

Reply:
agreed, I feel confuse too when I read the code, in this PR it actually just kept same behavior with original code.
let's change umempobj_settings_init() to make it match with the comment, but not sure if it will cause trouble in CI. will see...

Reply:
hmm, changing default to BMEM and hit a NLT failure (not very sure if it is related although), I'll change back to PMEM by default and remove the comment. To avoid CI failure.